### PR TITLE
Improve MDX range ignore support

### PR DESCRIPTION
**Pull request by nickrttn**
## Description

Currently, using [range ignore](https://prettier.io/docs/en/ignore.html#range-ignore) directives in JSX-style comments in MDX causes Prettier to [throw an error](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEwD0AqABABwE5wwwCWceAtMQOZQQHkDOMAhnjFhmgL4A6UfcAB4467SFCZYAKnEkBeLAAoAlFjkA+LMD5ZdBGAFc8ULAB4c6+E1NoLAbh1cHJ3aZmS06vn3TZ8hEjJKGjo4cgQAEw5uEAAaEAgcEmgGZFBWPAgAdwAFVgRUlGYAGyzmAE9U+IAjPGYwAGtCAGVmAFs4ABliKDhkADMShjgausaWnHqeqmQYPAMRkDg26rgIiLXO5igqA2YqOAAxOjbmImnkEGYDGAg4kAALGDbigHUH4itJsDhmgs-iAA3T7lS5gBhVEA9YZsHJ1KinAZDRYAKwYgma02KcAAigYIPAkcVhvFJngYZc2hFBPd8D0YK9iBEYA9kAAOAAMpMyw1edRwl38MMBfXiAEd8fA4YlClcGORems1vcCBLiAQ4ftEUhBsTFsM2sRZvN9VjcZK+jrkfEWNVGczWUgAEw2urEYrTADCEDa2qWDAArPcDMMpMxqoVdSSQICFgBJKAbWDNMB4YhJACCieaMHK2KJwy4XCAA).

This PR improves support for range ignore directives (eg. `prettier-ignore-start` and `prettier-ignore-end`) in JSX comments in MDX@2. JSX-style comment support [was recently added](https://github.com/prettier/prettier/pull/11563) to the MDX(/Markdown) parser by @wooorm. This PR builds on of that by amending the Markdown/MDX printer to make sure range ignored comments and the contained blocks are printed correctly.

While this makes sure range ignore directives in JSX-style comments work in MDX in most of the cases, there is one test case that is not fixed by this implementation:

```mdx
{/* prettier-ignore-start */}
export const Hello = () => {
   return (
     <p>Hello!<p/>);
}
{/* prettier-ignore-end */}
```

In this case, the last two lines of the function definition and the last range ignore comment are parsed as a single node:

```js
{
  type: 'jsx',
  value: '<p>Hello!<p/>);\n}\n{/* prettier-ignore-end */}',
  position: Position {
    start: { line: 7, column: 1, offset: 71 },
    end: { line: 9, column: 28, offset: 116 },
    indent: [ 1, 1 ]
  }
}
```

HTML comments _do_ work in this manner and I think that JSX and HTML comments should behave in a similar way. I don't know enough about the parser to fix this, and I suspect this will be parsed correctly when the mdx@2 parser is implemented in Prettier.

Please note that this is my first contribution to Prettier. Some stuff might be in the wrong place. Please let me know :)

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by jrson83** [comment]
Would be great if accepted.

**Comment by nickrttn** [comment]
I'll update it sometime this week to make sure tests are passing

**Review by wooorm — COMMENTED** [review]
One nit, otherwise good to go!

**Review by wooorm — COMMENTED** [review]
Looks good to me

/cc @fisker

**Review by wooorm — APPROVED** [review]

File: src/language-markdown/printer-markdown.js [hunk 1]
```diff
@@ -750,6 +750,21 @@ function printChildren(path, options, print, events = {}) {
...
 
+function printIgnoreComment(node) {
+  if (node.type === "html") {
+    return node.value;
+  }
+
+  if (
+    node.type === "paragraph" &&
+    Array.isArray(node.children) &&
+    node.children[0] &&
```
Comment (fisker) [inline src/language-markdown/printer-markdown.js:761]:
```suggestion
    node.children.length === 1 &&
```

To be safer?

**Review by fisker — APPROVED** [review]
Sorry, I missed this, LGTM. One comment.

**Comment by nickrttn** [comment]
@fisker updated and thank you!

**Comment by fisker** [comment]
@nickrttn The changlog didn't follow the template.

**Comment by nickrttn** [comment]
I'll update it to make sure it does

**Comment by nickrttn** [comment]
@fisker I've just updated the Changelog to follow the template. Hope this is ok, but please let me know if anything needs to change.

**Review by fisker — APPROVED** [review]

File: src/language-markdown/printer-markdown.js (hunk 1 above)
Comment (fisker) [inline src/language-markdown/printer-markdown.js:761]:
```suggestion
    node.children.length === 1 &&
```

**Comment by nickrttn** [comment]
@fisker is this ready to be merged?

**Comment by nickrttn** [comment]
@fisker Anything I can do to help you get this in?

**Comment by TxHawks** [comment]
This would be really helpful. Any chance to get this merged?

**Comment by B2o5T** [comment]
Please re-review to merge it!

**Review by wooorm — APPROVED** [review]

**Comment by B2o5T** [comment]
@fisker @kachkaev @sosukesuzuki ping

**Review by B2o5T — APPROVED** [review]

**Comment by 0xcadams** [comment]
We have a dependency on this PR for running prettier on our docs and would love to see this land! 👾

**Comment by fisker** [comment]
Sorry for the long delay.

**Comment by fisker** [comment]
Thanks for contributing! ❤️

**Pull request merged**